From 3e0ae8cc2d642f1af120fa7204c56d42280eb4d2 Mon Sep 17 00:00:00 2001
From: Dor Askayo <dor.askayo@gmail.com>
Date: Tue, 1 Feb 2022 21:10:41 +0200
Subject: [PATCH] Use $SOURCE_DATE_EPOCH for copyright year if defined

This allows reproducible sources for apps/progs.c and apps/progs.h.

This change is similar to 11d7d90344, but for apps/progs.pl.

CLA: trivial

Signed-off-by: Dor Askayo <dor.askayo@gmail.com>

diff --git a/apps/progs.pl b/apps/progs.pl
index 8a5759a961..77054902b4 100644
--- a/apps/progs.pl
+++ b/apps/progs.pl
@@ -21,7 +21,7 @@ die "Unrecognised option, must be -C or -H\n"
 my %commands     = ();
 my $cmdre        = qr/^\s*int\s+([a-z_][a-z0-9_]*)_main\(\s*int\s+argc\s*,/;
 my $apps_openssl = shift @ARGV;
-my $YEAR         = [localtime()]->[5] + 1900;
+my $YEAR         = [gmtime($ENV{SOURCE_DATE_EPOCH} || time())]->[5] + 1900;
 
 # because the program apps/openssl has object files as sources, and
 # they then have the corresponding C files as source, we need to chain
-- 
2.34.1

